networkit/centrality.pxd

```diff
@@ -1,27 +0,0 @@
-from libc.stdint cimport uint64_t
-from libcpp.vector cimport vector
-from libcpp.utility cimport pair
-from libcpp cimport bool as bool_t
-
-ctypedef uint64_t index
-ctypedef index node
-
-from .base cimport _Algorithm, Algorithm
-from .graph cimport _Graph, Graph
-from .structures cimport _Partition, Partition
-
-cdef extern from "<networkit/centrality/Centrality.hpp>":
-
-	cdef cppclass _Centrality "NetworKit::Centrality"(_Algorithm):
-		_Centrality(_Graph, bool_t, bool_t) except +
-		vector[double] scores() except +
-		vector[pair[node, double]] ranking() except +
-		double score(node) except +
-		double maximum() except +
-		double centralization() except +
-
-
-cdef class Centrality(Algorithm):
-	""" Abstract base class for centrality measures"""
-
-	cdef Graph _G
```

Comment:
Why is this deleted?

Reply:
`centrality.pxd` was not necessary because no class needed a `cimport`.

Reply:
I would suggest to restore this as otherwise other extension modules cannot export further centrality classes based on this base class. While there are none in NetworKit itself, this change allows providing extensions to NetworKit that can be distributed independently of NetworKit itself.

Reply:
I'm not sure I agree as this increases the amounts of files we have but do not actually need. I assume we would apply this uniformly to all C++ classes, and not just to centrality.

Reply:
The problem are only the Python classes that could be used as a base class as you cannot access the cdef-defined members in subclasses unless you can import their definition from a pxd file. Therefore, my argument primarily applies to all classes that are typically used as a base class and not all C++ classes. I think the number of classes this affects should be relatively low, maybe one or two per module. We can also do this later after merging this PR, this PR is already a huge improvement over the current state and should be merged as soon as possible in my opinion.